UM-USElab/gradecraft-development
# 2633 canvas api error response

### Status
**HOLD**

### Description
Gracefully handle any Canvas communication issues in the controller. This adds an exception handler block for any canvas call so the controller can redirect to a non-canvas controller method and display an alert to the user saying that the communication with Canvas did not work.

**WAITING ON WHAT TO DO ABOUT GETTING THE ACTUAL ERROR FROM CANVAS**

### Migrations
NO

### Steps to Test or Reproduce

1.  Login as an instructor
1. Link the Canvas course 97697 with any GC course
1. Import grades from assignment 103424
1. Should see error message and the error should be gracefully handled

### Impacted Areas in Application
List general components of the application that this PR will affect:

* Linking Canvas courses
* Canvas importing of assignments
* Canvas importing of grades

Closes #2633
